.github/workflows/gh-ci.yaml

```diff
@@ -15,165 +15,163 @@ defaults:
-          pip install asv
-        fi
-    
-    - name: check_setup
-      run: |
-        # Check OS and python setup
-        echo "OS: ${OS_NAME}"
-        which python
-        which pip
-        pip list
-        conda info
-        conda list
-
-    - name: install_hole
-      if : matrix.install_hole
-      run: |
-        # We manually build hole2 to avoid OS incompatibilities
-        git clone https://github.com/MDAnalysis/hole2.git
-        cd hole2/src
-        source ../source.apache
-        (make FC=${FC}) && (make PREFIX=${HOME}/hole2 FC=${FC} install)
-        source ../source.unset
-        echo "HOLE_BINDIR=${HOME}/hole2/bin" >> $GITHUB_ENV
-        echo "${HOME}/hole2/bin" >> $GITHUB_PATH
-
-    - name: install_mda
-      run: |
-        # TODO: using install instead of develop here causes coverage to drop
-        # for .pyx file. If possible a solution for this should be found.
-        (cd package/ && python setup.py develop) && (cd testsuite/ && python setup.py install)
-
-    - name: run_tests
-      if: contains(matrix.name, 'asv_check') != true
-      run: |
-        PYTEST_FLAGS="--disable-pytest-warnings --durations=50"
-        if [ ${{ matrix.codecov }} = "true" ]; then
-          PYTEST_FLAGS="${PYTEST_FLAGS} --cov=MDAnalysis --cov-report=xml"
-        fi
-        echo $PYTEST_FLAGS
-        pytest -n 2 testsuite/MDAnalysisTests $PYTEST_FLAGS
-
-    - name: run_asv
-      if: contains(matrix.name, 'asv_check')
-      run: |
-        cd benchmarks
-        time python -m asv check -E existing
-
-    - name: codecov
-      if: matrix.codecov
-      uses: codecov/codecov-action@v1
-      with:
-        file: coverage.xml
-        fail_ci_if_error: True
-        verbose: True
-
+      - uses: actions/checkout@v2
+
+      - name: setup_osx
+        if: startsWith(matrix.os, 'macOS')
+        run: |
+          # Set OS specific vars and compiler flags
+          echo "OS_NAME=osx" >> $GITHUB_ENV
+          # TODO: work out why this is necessary (from CI helpers)
+          echo "MACOSX_DEPLOYMENT_TARGET=10.9" >> $GITHUB_ENV
+          ulimit -S -n 2048
+          clang -v
+          echo "CC=clang" >> $GITHUB_ENV
+          clang++ -v
+          echo "CXX=clang++" >> $GITHUB_ENV
+          gfortran-9 -v
+          echo "FC=gfortran-9" >> $GITHUB_ENV
+
+      - name: setup_linux
+        if: startsWith(matrix.os, 'ubuntu')
+        run: |
+          # Set OS specific vars and compiler flags
+          echo "OS_NAME=linux" >> $GITHUB_ENV
+          gcc -v
+          echo "CC=gcc" >> $GITHUB_ENV
+          g++ -v
+          echo "CXX=g++" >> $GITHUB_ENV
+          gfortran -v
+          echo "FC=gfortran" >> $GITHUB_ENV
+
+      - name: setup_miniconda
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+          auto-update-conda: true
+          channel-priority: flexible
+          channels: conda-forge, biobuilds
```

Comment:
I'm glad that fixes it, it _should_ make maintaining the thing easier in the long run.

Probably something we need to look into in the long run (probably in that mamba PR I got open somewhere), is to see if having strict channel priority speeds up the setup time.

Reply:
As an extra thought, could you add a comment above this to clarify that the order here matters (with a reference to this PR)?